packages/vulcan-core/lib/modules/default_mutations.js

```diff
@@ -82,164 +83,197 @@ export const getDefaultMutations = (collectionName, options = {}) => {
+        const { selector, data } = input;
+        const collection = context[typeName];
 
         // check if document exists already
-        const existingDocument = await Connectors.get(collection, search, { fields: { _id: 1 } });
+        const existingDocument = await Connectors.get(collection, selector, { fields: { _id: 1 } });
 
         if (existingDocument) {
-          const editArgs = {
-            documentId: existingDocument._id,
-            set,
-            unset,
-          };
-          return await collection.options.mutations.edit.mutation(root, editArgs, context);
+          return await collection.options.mutations.update.mutation(root, { selector, data }, context);
         } else {
-          return await collection.options.mutations.new.mutation(root, { document: set }, context);
+          return await collection.options.mutations.create.mutation(root, { data }, context);
         }
       },
-    },
-
+    };
+  }
+  if (options.delete) {
     // mutation for removing a specific document (same checks as edit mutation)
 
-    remove: {
+    const deleteMutation = {
+      description: `Mutation for deleting a ${typeName} document`,
 
-      name: `${collectionName}Remove`,
-      
-      description: `Mutation for deleting a ${collectionName} document`,
-      
       check(user, document) {
         if (options.removeCheck) {
           return options.removeCheck(user, document);
         }
-        
+
         if (!user || !document) return false;
-        return Users.owns(user, document) ? Users.canDo(user, `${collectionName.toLowerCase()}.remove.own`) : Users.canDo(user, `${collectionName.toLowerCase()}.remove.all`);
+        return Users.owns(user, document)
+          ? Users.canDo(user, `${typeName.toLowerCase()}.remove.own`)
+          : Users.canDo(user, `${typeName.toLowerCase()}.remove.all`);
       },
-      
-      async mutation(root, {documentId}, context) {
 
-        const collection = context[collectionName];
+      async mutation(root, { input }, context) {
+        const { selector } = input;
+        const { documentId } = selector;
+        const collection = context[typeName];
 
         const document = await Connectors.get(collection, documentId);
         Utils.performCheck(this.check, context.currentUser, document, context);
 
-        return await removeMutator({
-          collection, 
-          documentId: documentId, 
+        return await deleteMutator({
+          collection,
+          documentId: documentId,
           currentUser: context.currentUser,
           validate: true,
           context,
         });
       },
+    };
 
-    },
+    mutations.delete = deleteMutation;
+    // OpenCRUD backwards compatibility
+    mutations.remove = deleteMutation;
   }
 
+  return mutations;
 };
 
-
 const registerCollectionCallbacks = collectionName => {
-
   collectionName = collectionName.toLowerCase();
 
   registerCallback({
-    name: `${collectionName}.new.validate`, 
```

Comment:
Should leave the backwards compatible callback names

Reply:
The callbacks are still there, `registerCallback` is only for the documentation.